### PR TITLE
Add clang-tidy in CI

### PR DESCRIPTION
**Pull request by innerout**
This PR adds clang-tidy linting in CI. It will check for missing braces around control statements since clang-format cannot detect it. It will fail in case of style errors.

### CODE REVIEW

File: applications/coreutils/mkdir.cpp [hunk 1]
```diff
@@ -28,9 +28,8 @@ static CommandLine cmdline = CMDLINE(
 void print_path(const char *path)
 {
     if (verbose)
-    {
```
Comment (Keyboard-Slayer) [inline applications/coreutils/mkdir.cpp:31]: Can you put those curly braces back please ?

**Review by Keyboard-Slayer — CHANGES_REQUESTED** [review]

File: applications/coreutils/mkdir.cpp (hunk 1 above)
Comment (innerout) [reply applications/coreutils/mkdir.cpp:31]: I have removed them to test that clang-tidy works as it should. When I make it work I will undo this change.

Comment (Keyboard-Slayer) [reply applications/coreutils/mkdir.cpp:31]: Ok no problem, You should write in on the description of the pr :sweat_smile:

**Comment by innerout** [comment]
You can merge this PR. @sleepy-monax @Keyboard-Slayer

File: .github/workflows/check.yml [hunk 2]
```diff
@@ -55,3 +55,22 @@ jobs:
...
+
+  codestyle:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Install packages
+      run: |
+        sudo apt install -y nasm gcc g++ gcc-multilib g++-multilib make binutils grub-pc-bin qemu-system-x86 clang-tidy
```
Comment (Keyboard-Slayer) [inline .github/workflows/check.yml:67]: Why do need all those deps ?

**Review by Keyboard-Slayer — CHANGES_REQUESTED** [review]

File: .github/workflows/check.yml (hunk 2 above)
Comment (innerout) [reply .github/workflows/check.yml:67]: Not all of them are useful, I was trying to install a missing header and I thought one of these packages would install it.
I will remove the unneeded ones.

**Comment by innerout** [comment]
ping @Keyboard-Slayer

File: .github/workflows/check.yml [hunk 3]
```diff
@@ -64,9 +64,7 @@ jobs:
 
     - name: Install packages
       run: |
-        sudo apt install -y nasm gcc g++ gcc-multilib g++-multilib make binutils grub-pc-bin qemu-system-x86 clang-tidy
-        sudo apt install -y build-essential bison flex libgmp3-dev libmpc-dev
-        sudo apt install -y xorriso mtools inkscape libmpfr-dev texinfo libfuse-dev
+        sudo apt install -y nasm gcc gcc-multilib clang-tidy
```
Comment (Keyboard-Slayer) [inline .github/workflows/check.yml:67]: Heuuu you just need `clang-tidy` :sweat_smile:  (the rest is already installed above)

**Review by Keyboard-Slayer — CHANGES_REQUESTED** [review]

File: .github/workflows/check.yml (hunk 3 above)
Comment (innerout) [reply .github/workflows/check.yml]: I need also gcc-multilib in order for clang-tidy not to report some missing headers.

File: .github/workflows/check.yml [hunk 4]
```diff
@@ -64,9 +64,7 @@ jobs:
 
     - name: Install packages
       run: |
-        sudo apt install -y nasm gcc g++ gcc-multilib g++-multilib make binutils grub-pc-bin qemu-system-x86 clang-tidy
-        sudo apt install -y build-essential bison flex libgmp3-dev libmpc-dev
-        sudo apt install -y xorriso mtools inkscape libmpfr-dev texinfo libfuse-dev
+        sudo apt install -y gcc-multilib clang-tidy
```
Comment (Keyboard-Slayer) [inline .github/workflows/check.yml:67]: Do you really need `gcc-multilib` ??

Comment (innerout) [reply .github/workflows/check.yml:67]: Yes, for some reason in the CI clang-tidy does not find a header related to bitwise operations, and the only way to fix it was to install `gcc-multilib`.

Comment (Keyboard-Slayer) [reply .github/workflows/check.yml:67]: weird :thinking:

**Review by Keyboard-Slayer — APPROVED** [review]

**Pull request merged**